internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlyStoreConfigRepository.java

```diff
@@ -131,63 +130,63 @@ public StoreConfig getStoreConfigOrThrow(String storeName) {
     return storeConfig.get();
   }
 
-  @Override
-  public List<StoreConfig> getAllStoreConfigs() {
-    return new ArrayList<>(storeConfigMap.get().values());
-  }
-
-  protected StoreConfigAddedOrDeletedChangedListener getStoreConfigAddedOrDeletedListener() {
+  @VisibleForTesting
+  StoreConfigAddedOrDeletedChangedListener getStoreConfigAddedOrDeletedListener() {
     return storeConfigAddedOrDeletedListener;
   }
 
-  protected StoreConfigChangedListener getStoreConfigChangedListener() {
+  @VisibleForTesting
+  StoreConfigChangedListener getStoreConfigChangedListener() {
     return storeConfigChangedListener;
   }
 
+  @VisibleForTesting
+  Set<String> getAvailableStoreSet() {
+    return availableStoreSet.get();
+  }
+
+  @VisibleForTesting
+  Map<String, StoreConfig> getLoadedStoreConfigMap() {
+    return loadedStoreConfigMap;
+  }
+
   protected class StoreConfigAddedOrDeletedChangedListener implements IZkChildListener {
     @Override
-    public void handleChildChange(String parentPath, List<String> currentChildren) throws Exception {
-      synchronized (storeConfigMap) {
-        Map<String, StoreConfig> map = new HashMap<>(storeConfigMap.get());
-        List<String> newStores =
-            currentChildren.stream().filter(newStore -> !map.containsKey(newStore)).collect(Collectors.toList());
-
-        Set<String> deletedStores = new HashSet<>(map.keySet());
-        currentChildren.forEach(deletedStores::remove);
+    public void handleChildChange(String parentPath, List<String> currentChildren) {
+      synchronized (availableStoreSet) {
+        Set<String> storeSetSnapshot = new HashSet<>(availableStoreSet.get());
+        List<String> newStores = currentChildren.stream()
```

Comment:
Do we have this variable at all?

Reply:
It's purely for information purpose and has some value. I will make it a `int`

Reply:
Updated.

Reply:
@gaojieliu  can you please take another look when you have time?